### ISSUE TITLE
Request: Add ability to share and link from other apps

### ISSUE THREAD

**Issue by norrinr** [junjunguo/PocketMaps#62]
It would be nice if I could share a location with PocketMaps. I realize that it wouldn't do an address yet, but maybe coordinates from another map site page.

Also, it might be a geo: link, but when I use AddressToGPS and it finds some coordinates for me, there's an option to "navigate to" the location. PocketMaps does not come up in the list to open the link with, but ZANavi does. It would be nice if it opened up the location in PocketMaps and pre-filled the nav start/dest fields with current location (if avail) and the location that was shared/linked.

**Comment by glitchphan** [comment]
This. Glad someone else wanted this too!

**Comment by glitchphan** [comment]
Closed just because old?

**Comment by Starcommander** [comment]
I closed all old issues, because since 2016 there are new versions of pocketmaps.

Does anyone know more about such a geo-link and how does this work technically?

Note, that there are actually 2 pocketmaps apps on google-play, so you have maybe to reinstall the app to get version 2.0 (starcommander).
Or you can install the app from fdroid.

**Comment by Starcommander** [comment]
It seems to have to do with RFC5870
[https://en.wikipedia.org/wiki/Geo_URI_scheme](url)

**Comment by Starcommander** [comment]
For the manifest:

```
<intent-filter>
    <action android:name="android.intent.action.VIEW"/>
    <category android:name="android.intent.category.DEFAULT"/>
    <category android:name="android.intent.category.BROWSABLE"/>
    <data android:scheme="geo"/>
</intent-filter>
```

**Comment by Starcommander** [comment]
There is also a google.navigation scheme:
[https://stackoverflow.com/questions/4369012/android-intent-filter-for-geo-uri#4392898](url)
[https://github.com/osmandapp/Osmand/blob/master/OsmAnd/AndroidManifest.xml](url)
- To get the Url: Intent.getData()
- Possible problems: Select correct map first?

**Comment by natrius** [comment]
I would like to add: Use the adress from the adress-book. If i integrate an adress there for a contact, i would like to find a way fast. 
It seems like its complicated to integrate due different adress-schemes in different countrys. 

To deal with that, my suggestions would be:

1. set a format in advance. "If you want to use this feature, the adresses in your contacts has to be in this format: ...."
2. integrate a choice. Like "I'm using format A or B or C"
3. Integrate a possiblity to integrate it by yourself. By dragging around "Number", "Street", "Postal", "City" and so on. 

And everybody can choose if he wants to do this or not.

**Comment by Starcommander** [comment]
Available in next version 2.3

**Comment by glitchphan** [comment]
Tried it on 2.7. Didn't work with message Error getting input: input uri has no parameters.


-------- Original Message --------

**Comment by glitchphan** [comment]
Tried it with version 2.7 from F-Droid and it said Error getting input: input uri has no parameters!

Trying to click from AddressToGPS (Simple address/POI search) - https://f-droid.org/app/me.danielbarnett.addresstogps

**Comment by Starcommander** [comment]
It works at least with calendar app on my device.
I will try it with your AddressToGps-App as I have the time.

**Comment by Starcommander** [comment]
Unfortunately the app AddressToGps does not work anymore because of this issue:
https://github.com/DanielBarnett714/AddressToGPS/issues/13
So I can not test the behaviour.
But it is possible to search a location via pocketmaps directly.

**Comment by Starcommander** [comment]
I was able to reproduce the issue.
Just select a geo-link from the Website inside of the Android Web-Browser:
https://en.wikipedia.org/wiki/Geo_URI_scheme

**Comment by Starcommander** [comment]
Fixed in V2.8

**Comment by glitchphan** [comment]
After upgrading to 2.9 from F-Droid, I can send locations from AddressToGPS II. Thank you!
